clients/src/test/java/org/apache/kafka/clients/producer/internals/RecordAccumulatorTest.java

```diff
@@ -81,15 +81,15 @@ public void testFull() throws Exception {
         int appends = batchSize / msgSize;
         for (int i = 0; i < appends; i++) {
             // append to the first batch
-            accum.append(tp1, 0L, key, value, null, maxBlockTimeMs);
+            accum.append(tp1, 0L, key, value, -1L, null, maxBlockTimeMs);
```

Comment:
maybe in the PR that actually gets opened against Kafka, but I wasn't focusing on writing clean code for this spike.  The goal here was just to get conditional append working and be able to provide an estimate for how much work it would be to do this and get it accepted upstream.